### PR TITLE
[StructuralMechanics] correct visualization-process with constraints

### PR DESCRIPTION
**Pull request by philbucher**
without this the animation on slave-dofs does not work

Examples with and without this fix (click to enlarge); the constraints are applied inbetween the individual components

| With consideration of Constraints:  | Without consideration of Constraints: |
| ------------- | ------------- |
| with_constraints_1 | without_constraints_1  |
| with_constraints_2  | without_constraints_2  |

**NOTE:** this requires #4159 , it won't compile as is!

### CODE REVIEW

**Review by loumalouomega — APPROVED** [review]
Approving

**Comment by philbucher** [comment]
now it should work, can you reapprove pls?

File: applications/StructuralMechanicsApplication/custom_processes/postprocess_eigenvalues_process.cpp [hunk 1]
```diff
@@ -106,8 +107,7 @@ namespace Kratos
             {
                 label = GetLabel(j, eigenvalue_vector[j]);
 
-                for(auto& r_node : mrModelPart.Nodes())
-                {
+                for (auto& r_node : mrModelPart.Nodes()) {
```
Comment (RiccardoRossi) [inline applications/StructuralMechanicsApplication/custom_processes/postprocess_eigenvalues_process.cpp:110]: you may want to consider doing this in OpenMp (in a future PR)

File: applications/StructuralMechanicsApplication/custom_processes/postprocess_eigenvalues_process.cpp [hunk 2]
```diff
@@ -118,6 +118,12 @@ namespace Kratos
                     SizeType k = 0;
                     for (auto& r_dof : r_node_dofs)
                         r_dof.GetSolutionStepValue(0) = std::cos(angle) * r_node_eigenvectors(j,k++);
```
Comment (RiccardoRossi) [inline applications/StructuralMechanicsApplication/custom_processes/postprocess_eigenvalues_process.cpp:120]: also i would move the std::cos out of the loop

**Review by RiccardoRossi — APPROVED** [review]
I am approving so that it can make it for the release, but please consider my comments

File: applications/StructuralMechanicsApplication/custom_processes/postprocess_eigenvalues_process.cpp (hunk 2 above)
Comment (philbucher) [reply applications/StructuralMechanicsApplication/custom_processes/postprocess_eigenvalues_process.cpp:120]: ok makes sense, thx for the suggestions

**Pull request merged**